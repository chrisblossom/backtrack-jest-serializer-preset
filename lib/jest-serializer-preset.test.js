'use strict';

const {
	transformConfig,
} = require('@backtrack/core/dist/options-file/transform-config');
const { test: testFn } = require('./jest-serializer-preset');

describe('serializer', () => {
	test('removes hash from files dest object', () => {
		const backtrackConfig = {
			packageJson: { outside: true },
			files: [
				{ src: '__sandbox__/file1.js', dest: 'package.json' },
				{ src: '__sandbox__/file2.js', dest: '.babelrc.js' },
			],
		};

		const result = transformConfig(backtrackConfig, __dirname);

		expect(result).toMatchSnapshot();
	});
});

describe('test', () => {
	const baseFiles = {
		files: [
			'package.json',
			'.babelrc.js',
		],
		absolute: {
			'package.json': 'backtrack-jest-serializer-preset/package.json',
			'.babelrc.js': 'backtrack-jest-serializer-preset/.babelrc.js',
		},
		hash: {
			'package.json':
				'8ecbd0ea1db68b7e5e44874b579d104a1d65d654276dcf089738018edab680a0',
			'.babelrc.js': '',
		},
		allowChanges: { 'package.json': false, '.babelrc.js': false },
		ignoreUpdates: { 'package.json': false, '.babelrc.js': false },
	};

	test('handles dest files', () => {
		const result = testFn(baseFiles);

		expect(result).toEqual(true);
	});

	test('handles null obj', () => {
		const obj = null;

		const result = testFn(obj);

		expect(result).toEqual(false);
	});

	test('handles null files', () => {
		const obj = {
			...baseFiles,
			files: null,
		};

		const result = testFn(obj);

		expect(result).toEqual(false);
	});

	test('handles string files', () => {
		const obj = {
			...baseFiles,
			files: 'some string',
		};

		const result = testFn(obj);

		expect(result).toEqual(false);
	});

	test('handles plain object files', () => {
		const obj = {
			...baseFiles,
			files: {},
		};

		const result = testFn(obj);

		expect(result).toEqual(false);
	});
});
