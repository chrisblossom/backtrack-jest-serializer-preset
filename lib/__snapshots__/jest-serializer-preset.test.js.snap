// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer removes hash from files dest object 1`] = `
{
  "files": {
    "dest": {
      "absolute": {
        ".babelrc.js": "<PROJECT_ROOT>/.babelrc.js",
        "package.json": "<PROJECT_ROOT>/package.json",
      },
      "allowChanges": {
        ".babelrc.js": false,
        "package.json": false,
      },
      "files": [
        "package.json",
        ".babelrc.js",
      ],
      "ignoreUpdates": {
        ".babelrc.js": false,
        "package.json": false,
      },
    },
    "makeDirs": [
      "dist",
      "src",
    ],
    "src": {
      "absolute": {
        "lib/__sandbox__/file1.js": "<PROJECT_ROOT>/lib/__sandbox__/file1.js",
        "lib/__sandbox__/file2.js": "<PROJECT_ROOT>/lib/__sandbox__/file2.js",
      },
      "files": [
        "lib/__sandbox__/file1.js",
        "lib/__sandbox__/file2.js",
      ],
      "hash": {
        "lib/__sandbox__/file1.js": "49598ea59816da0629609bb00775082fb21490af94fbcc8fbbbb120a6f5dc135",
        "lib/__sandbox__/file2.js": "6cfc52d273559d0815ab3c56e1a1ad59b540948ddff9162991599c068d55e46a",
      },
    },
  },
  "packageJson": [
    {
      "outside": true,
    },
  ],
}
`;
